ADAL/src/ADAuthenticationParameters+Internal.m

```diff
@@ -95,79 +96,175 @@ + (ADAuthenticationError *)invalidHeader:(NSString *)headerContents
 + (NSDictionary *)extractChallengeParameters:(NSString *)headerContents
                                        error:(ADAuthenticationError * __autoreleasing *)error;
 {
-    NSError* rgError = nil;
-    __block ADAuthenticationError* adError = nil;
+    NSMutableArray *items = [self extractItems:headerContents];
     
-    if ([NSString adIsStringNilOrBlank:headerContents])
+    NSInteger bearerStartIndex = NSNotFound;
+    for (int i = 0; i < items.count; i++)
     {
-        adError = [self invalidHeader:headerContents];
+        NSString *item = items[i];
+        NSRange range = [item rangeOfString:BearerWithParamExpression options:NSRegularExpressionSearch];
+        if (range.location != NSNotFound)
+        {
+            // Remove 'Bearer'.
+            NSRegularExpression *regex = [NSRegularExpression regularExpressionWithPattern:BearerSchemaNameExpression options:NSRegularExpressionCaseInsensitive error:nil];
+            NSString *param = [regex stringByReplacingMatchesInString:item options:0 range:NSMakeRange(0, [item length]) withTemplate:@""];
+            
+            // Save cleared param.
+            items[i] = param;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this hazardous cast. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=objc%3AS856)